### DISCUSSION TITLE
Build components that depend on pinia in library mode

### DISCUSSION THREAD

**Discussion by leopsidom** [vuejs/pinia#1059] [Help and Questions] [answered]
I'm trying to build a component that depend on pinia to a npm package so I can import from another application. However currently I'm getting this error:

> getActivePinia was called with no active Pinia. Did you forget to install pinia?
> const pinia = createPinia()
> app.use(pinia)

Even though I have already installed pinia in my main application. 

Is this the expected behavior ? Though I have a feeling it might not be a good practice to include pinia or states into a library, I wanted to check if anyone else has encountered this problem.

**Comment by Nerderer** [comment]
Did you solve this?

We have the same issue, where we have component library (that uses Pinia) and a main application that consumes the components (this application is also using Pinia). 

I can make it work by passing the active Pinia instance into an initialize components function that we use, then use `setActivePinia(pinia)` here. But I get this warning in the log:

`[Vue warn]: injection "Symbol(pinia)" not found. `

**Comment by krckyboy** [comment]
Bump.

**Comment by posva** [comment, marked as answer]
Make sure to:

- externalise pinia from the lib build
- Use one single pinia in the application

**Reply by mrichardsonjr** [reply]
"externalise pinia from the lib build"  -- finally fixed the issue that's been driving me nuts.
God bless you, @posva

**Reply by botmaster** [reply]
Hello,
Not sure I understand. How to externalise pinia ?

**Reply by posva** [reply]
it depends on what you use to bundle your lib. You should be able to find the exact option with the word "external" or "exclude". This is for tsup: https://tsup.egoist.dev/#excluding-packages. For vite, `rollupOptions.external` should do

**Reply by botmaster** [reply]
Ok thank you @posva !

**Reply by derwaldgeist** [reply]
I have externalised pinia in the library and am using just one pinia in the app. But still I am getting this error:

> pinia.mjs:1735 Uncaught (in promise) Error: [🍍]: "getActivePinia()" was called but there was no active Pinia. Are you trying to use a store before calling "app.use(pinia)"?

It happens when the library component tries to call useMyStore(). (defineStore just works fine.)

I checked the Vue app context in both the app and the component (before I call useMyStore()), and both seem to be the same.

What might be the issue why useMyStore() still cannot find the pinia instance of the hosting app?
